Add launch.json

tzjames opened the pull request:
### Features and Changes

Allow for debugging with VS code.  The backend debugger will restart on changes.

There is also a front-end debugger, but it seems less than useful as it is, as a breakpoint has to be added within google chrome.  When it is hit a read-only version of the file is opened in VS-code.  This doesn't seem to get us much more than just debugging straight in chrome.  I tried briefly getting the front-end to create source maps, but it didn't seem to work.  Still adding it for completeness and it does at least allow both the front-end and back-end to start with one click.

### Testing

In VS code click on Run and Debug or ( Shift - Apple - D)
From the drop down menu select Debug Fullstack.
Click start.
Set a breakpoint in both the front end and back end.
Hit a url that will trigger the breakpoint.  
See it stop.
Edit a file in a backend.
See the backend restart.
Hit a url that will trigger the breakpoint.
See that the backend will stop at the correct line numbers given the edits, because it had successfully restarted.

tinahollygb reviewed (approved):
I tested it out. The developer experience is a bit cumbersome. It seems "Debug Back-end" on its own is not usable, which is what I tried initially, and "Debug Full-stack" opens an incognito browser. It did break if I ran "Debug Full-stack" but not just "Debug Back-end." It did not break on anything in the front-end but I didn't add breakpoints to the Chrome dev tools, only VSCode.

It also doesn't seem to break at all backend endpoints. It worked fine for the feature toggle endpoint but it didn't trigger for the health check endpoints (`"/"` and `"/healthcheck"`).

I think if you find this helpful it makes sense to add it but I'm not sure if it alludes to more functionality than it provides. Still I think it makes sense to add it since it does work in some cases.

tzjames commented:
> I tested it out. The developer experience is a bit cumbersome. It seems "Debug Back-end" on its own is not usable, which is what I tried initially, and "Debug Full-stack" opens an incognito browser. It did break if I ran "Debug Full-stack" but not just "Debug Back-end." It did not break on anything in the front-end but I didn't add breakpoints to the Chrome dev tools, only VSCode.

Yes I agree it is a bit clunky to use.  As I said in the PR request the front end break points have to be added within the chrome that opens up, which can be done anyway.  So it is only really useful in that then Debug Full stack can be done without running a separate `yarn dev` in projects/frontend.

> It also doesn't seem to break at all backend endpoints. It worked fine for the feature toggle endpoint but it didn't trigger for the health check endpoints ("/" and "/healthcheck").
@tinahollygb 
I added a breakpoint at app.ts:138  for / and it worked fine.
I added a breakpoint at app.ts:125 for /healthcheck and it also worked fine.
Where did you add yours?

I'll go ahead and land this.  It is possible that the front-end can be improved but I don't think it is worth much more of my time at this point.
